de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/ASwtBotTestCase.java

```diff
@@ -336,18 +330,17 @@ protected IProject createProjectWithConcepts(String projectName) {
 	 *
 	 */
 	protected void propagateInheritance() {
-		bot.saveAllEditors();
+		bot.waitUntil(new VirSatWaitForProjectBuilder(1));
 		waitForEditingDomainAndUiThread();
 		bot.waitUntil(Conditions.waitForJobs(ResourcesPlugin.FAMILY_AUTO_BUILD, "eclipse auto builders (inheritance builder)"));
 	}
+	
 	/**
 	 * waits for calculation builder 
 	 *
 	 */
 	protected void waitCalculationBuilder() {
-		bot.saveAllEditors();
-		waitForEditingDomainAndUiThread();
-		bot.waitUntil(Conditions.waitForJobs(ResourcesPlugin.FAMILY_AUTO_BUILD, "eclipse auto builders (calculation builder)"));
+		propagateInheritance();

```

Comment:
waitForCalculationBuilder() calling propagateInheritance() looks weird.
Either something is weird in the logic or the names should reflect the logic changes that happened.

Reply:
Right removed method and renamed one stating to wait for all builders and ui thread.